### PR TITLE
#1206 Add an option to enable the dynawoexec file

### PR DESCRIPTION
**Pull request by rosiereflo**

### CODE REVIEW

File: dynawo/sources/Simulation/DYNSimulation.h [hunk 1]
```diff
@@ -533,6 +533,9 @@ class Simulation {
   exportTimelineMode_t exportTimelineMode_;  ///< timeline's output mode
   std::string timelineOutputFile_;  ///< timeline's export file
 
+  std::string timestepsOutputFile_;  ///< timesteps's export file
+  int timestepsSteps_;  ///< timesteps's steps
```
Comment (adriguir) [inline dynawo/sources/Simulation/DYNSimulation.h:537]: timesteps' steps and timesteps export file

**Review by adriguir — CHANGES_REQUESTED** [review]
An update in the documentation would be nice too, as well as perhaps one example using the structure in the examples or nrt test cases.

I find the name "timesteps" for the output category a bit miscellaneous as it doesn't really output the used timesteps but the current time. I don't really know if "currentTime" is a better option, but I don't have any other clear name coming to my mind thus open question. Perhaps something in relation to progress? @thiebarrrte @gautierbureau @RomainLosseauRTE any suggestion?

**Comment by rosiereflo** [comment]
I named it like this as at the end of the simulation we dumped the time steps into this file, but I am open to any better suggestion!

**Comment by adriguir** [comment]
> I named it like this as at the end of the simulation we dumped the time steps into this file, but I am open to any better suggestion!

I think time step itself is the source of confusion. It could represent both the h and the t (in t(n+1) = t(n) + h(n)) so I think it depends what comes to your mind first (for me it's h but this is perhaps personal). Let's see if someone else has a proposal that convinces everyone!

File: dynawo/sources/Simulation/DYNSimulation.cpp [hunk 2]
```diff
@@ -337,6 +339,22 @@ Simulation::configureTimelineOutputs() {
   }
 }
 
+void
+Simulation::configureTimestepsOutputs() {
+  // Timeline settings
```
Comment (gautierbureau) [inline dynawo/sources/Simulation/DYNSimulation.cpp:344]: wrong comment

**Review by gautierbureau — APPROVED** [review]

**Comment by gautierbureau** [comment]
> > I named it like this as at the end of the simulation we dumped the time steps into this file, but I am open to any better suggestion!
> 
> I think time step itself is the source of confusion. It could represent both the h and the t (in t(n+1) = t(n) + h(n)) so I think it depends what comes to your mind first (for me it's h but this is perhaps personal). Let's see if someone else has a proposal that convinces everyone!

From a numerical analysis point of view time-step will always be h and not the time itself so I agree with the remark if we print the time and not the time-step we should use something else.

**Comment by rosiereflo** [comment]
What do you think of timetable?

**Comment by adriguir** [comment]
> What do you think of timetable?

Fine for me.

**Comment by gautierbureau** [comment]
> What do you think of timetable?

ok for me also

**Comment by rosiereflo** [comment]
I modified the name and added an example and a description in the documentation

File: dynawo/sources/Simulation/DYNSimulation.cpp (hunk 2 above)
Comment (rosiereflo) [reply dynawo/sources/Simulation/DYNSimulation.cpp]: solved

File: dynawo/sources/Simulation/DYNSimulation.h (hunk 1 above)
Comment (rosiereflo) [reply dynawo/sources/Simulation/DYNSimulation.h]: solved

**Review by adriguir — APPROVED** [review]

**Pull request merged**